db/indextest/indextest_test.go

```diff
@@ -310,15 +317,42 @@ func TestInitializeIndexesConcurrentMultiNode(t *testing.T) {
 
 	bucket := base.GetTestBucket(t)
 	defer bucket.Close(base.TestCtx(t))
-
 	var wg sync.WaitGroup
 	wg.Add(numSGNodes)
 	for i := 0; i < numSGNodes; i++ {
 		ctx := base.CorrelationIDLogCtx(context.Background(), fmt.Sprintf("test-node-%d", i))
 		go func() {
 			defer wg.Done()
-			setupN1QLStore(ctx, t, bucket, false)
+			setupN1QLStore(ctx, t, bucket, false, db.DefaultNumIndexPartitions)
 		}()
 	}
 	wg.Wait()
 }
+
+func TestPartitionedIndexes(t *testing.T) {
+	if !base.TestUseXattrs() {
+		t.Skip("TestPartitionedIndexes only works with UseXattrs=true")
+	}
+	numPartitions := uint32(8)
+	serverless := false
+
+	database, ctx := db.SetupTestDBWithOptions(t, db.DatabaseContextOptions{NumIndexPartitions: base.Ptr(numPartitions)})
+	defer database.Close(ctx)
+
+	setupN1QLStore(ctx, t, database.Bucket, serverless, numPartitions)
+	gocbBucket, err := base.AsGocbV2Bucket(database.Bucket)
+	require.NoError(t, err)
+	for _, dsName := range []sgbucket.DataStoreName{db.GetSingleDatabaseCollection(t, database.DatabaseContext).GetCollectionDatastore(), database.MetadataStore} {
+		allIndexes, err := gocbBucket.GetCluster().Bucket(gocbBucket.BucketName()).Scope(dsName.ScopeName()).Collection(dsName.CollectionName()).QueryIndexes().GetAllIndexes(nil)
+		require.NoError(t, err)
+		for _, index := range allIndexes {
+			if strings.HasPrefix(index.Name, "sg_allDocs") || strings.HasPrefix(index.Name, "sg_channels") {
+				require.True(t, strings.HasSuffix(index.Name, "x1_p8"), "expected 8 partitions for %+v", index)
```

Comment:
Is it possible to assert on the _actual_ number of partitions using this API? I can't tell whether it will be in `index.Params`

If so, changing `numPartitions` to something other than 8 and doing that assertion seems like an easy enhancement to this test to cover the options on query creation - likewise for the else case below - ensuring we only have 1 partition for everything else